.github/workflows/test.yml

```diff
@@ -60,12 +60,14 @@ jobs:
           fetch-depth: 0
           persist-credentials: false
       - name: Build Thrust tests
-        run: bash ./ci/build_thrust.sh
+        run: bash ./ci/build_thrust.sh 
+      - name: tar Thrust build artifacts
+        run: tar czf thrust-build-artifacts.tar.gz ./thrust
       - name: Upload Thrust build artifacts
         uses: actions/upload-artifact@v3
         with:
           name: thrust-build-artifacts
-          path: ./thrust/
+          path: thrust-build-artifacts.tar.gz
```

Comment:
Do we really want to store build artifacts?

That will be a massive waste of storage space and most likely soon lead to some overflows

Reply:
The storage is free and not managed by us.

If we don't do this, then we'd have to build tests twice and given how long our tests take to compile, that would be disastrous.